src/inc/quic_cert.h

```diff
@@ -132,3 +142,33 @@ CxPlatCertVerify(
         const uint8_t *Signature,
     _In_ size_t SignatureLength
     );
+
+
+//
+// Convert platform specific certificate to requested format.
+// OutputBuffer is allocated and needs to be freed with CxPlatQuicBufferFree.
+//
+_Success_(return != FALSE)
+BOOLEAN
+CxPlatCertConvert(
+    _In_ QUIC_CERTIFICATE* Certificate,
+    _In_ QUIC_CERTIFICATE_FORMAT OutputFormat,
+    _Inout_ QUIC_BUFFER* OutputBuffer);
+
+//
+// Convert platform specific certificate chain to requested format.
+// OutputBuffer is allocated and needs to be freed with CxPlatQuicBufferFree.
+//
+_Success_(return != FALSE)
+BOOLEAN
+CxPlatCertChainConvert(
+    _In_ QUIC_CERTIFICATE_CHAIN* CertificateChain,
+    _In_ QUIC_CERTIFICATE_FORMAT OutputFormat,
+    _Inout_ QUIC_BUFFER* OutputBuffer);
+
+
+//
+// This will free QUIC_BUFFER resources if allocated by MsQiuc.
+// This has undefined and possibly dangerous behavior if Buffer was not allocated by MsQuic.
+//
+void CxPlatQuicBufferFree(_In_ QUIC_BUFFER* Buffer);
```

Comment:
We only expose functions via the API table (in msquic.h). So these will need to be defined in there. I'd also prefer to return `QUIC_STATUS` instead of `BOOLEAN`.